sqlglot/dialects/dialect.py

```diff
@@ -117,20 +119,50 @@ def __eq__(cls, other: t.Any) -> bool:
     def __hash__(cls) -> int:
         return hash(cls.__name__.lower())
 
+    def __getattr__(cls, name):
```

Comment:
Should this be a `@property`? Or is there some metaclass complexity?

Reply:
That's what I tried first, but when you do `metaclass.some_property`, it returns a `property` object instead of the actual value... Python 🤷‍♂️

```python
>>> class Meta(type):
...     @property
...     def description(cls):
...         return f"This is a metaclass named {cls.__name__}"
...
>>> class MyClass(metaclass=Meta):
...     pass
...
>>> print(Meta.description)  # This is a metaclass named Meta
<property object at 0x104877100>
>>>
>>> print(MyClass.description)
This is a metaclass named MyClass
```

Reply:
🤷 I can't imagine anyone is referencing the metaclass directly? They definitely _shouldn't_.

But all this metaclass stuff is messy anyways, so I'm happy with whatever

Reply:
Ack, changed my mind and will refactor in favor of `property`. Thanks for poking here.